guides/source/active_record_migrations.md

```diff
@@ -482,34 +617,38 @@ change_column :products, :part_number, :text
 
 This changes the column `part_number` on products table to be a `:text` field.
 
-NOTE: The `change_column` command is **irreversible**.
-You should provide your own `reversible` migration, like we discussed
-[before](#making-the-irreversible-possible).
+NOTE: The `change_column` command is **irreversible**. To ensure your migration
+can be safely reverted, you will need to provide your own `reversible`
+migration. You can read more about reversible migrations
+[here](#using-reversible).
 
-Besides `change_column`, the [`change_column_null`][] and [`change_column_default`][]
-methods are used specifically to change a null constraint and default values of
-a column.
+Besides `change_column`, the [`change_column_null`][] and
+[`change_column_default`][] methods are used to change a null constraint and
+default values of a column.
 
 ```ruby
 change_column_null :products, :name, false
 change_column_default :products, :approved, from: true, to: false
 ```
 
-This sets `:name` field on products to a `NOT NULL` column and the default
-value of the `:approved` field from true to false. Both of these changes will
-only be applied to future transactions, any existing records do not apply.
+This sets `:name` field on products to a `NOT NULL` column and the default value
+of the `:approved` field from true to false. Both of these changes will only be
```

Comment:
Hm. If a column has null values in it, and we add a "not null" constraint, then the not null constraint won't apply, because existing rows would violate the constraint. We'd have to remove the rows or set the columns to non-null values for the constraint to apply.

For that reason I wouldn't include the not null addition in being relevant to only "only new rows" since existing rows are checked.